BraveShareTo/ShareToBraveViewController.swift

```diff
@@ -35,21 +39,24 @@ class ShareToBraveViewController: SLComposeServiceViewController {
         
         provider.loadItem(of: provider.isUrl ? kUTTypeURL : kUTTypeText) { item, error in
             var urlItem: URL?
+            var nonUrlText: String?
```

Comment:
This work, but I think we can make the code cleaner now, if you look at logic below that, there is a lot of nested ifs and the code is not super readable,

For us differences between text and url are small, we can remove code duplication a little and improve architecture

What do you think if we create a new structure to `ShareToBraveViewController` class holding either URL or text and handling the scheme properly,
Sample code

```private struct Scheme {
        private enum SchemeType {
            case url, query
        }
        
        private let type: SchemeType
        private let urlOrQuery: String
        
        init?(item: NSSecureCoding) {
            if let text = item as? String {
                urlOrQuery = text
                type = .query
            } else if let url = (item as? URL)?.absoluteString.firstURL?.absoluteString {
                urlOrQuery = url
                type = .url
            } else {
                return nil
            }
        }
        
        var schemeUrl: URL? {
            guard let percentEncodedUrlOrQuery = urlOrQuery
                    .addingPercentEncoding(withAllowedCharacters: .alphanumerics) else {
                return nil
            }
            
            // Can improve even further with adding param using `URLComponents`
            switch type {
            case .url:
                return URL(string: "brave://open-url?url=\(percentEncodedUrlOrQuery)")
            case .query:
                return URL(string: "brave://search?q=\(percentEncodedUrlOrQuery)")
            }
        }
    }
```

Then calling it from `provider.loadItem` is much cleaner, you can simply do:

```
guard let item = item, let schemeUrl = Scheme(item: item)?.schemeUrl else {
    self.cancel()
    return
 }
            
self.handleUrl(schemeUrl)
```

what do you think about it?